python/lsst/meas/astrom/utils.cc

```diff
@@ -53,282 +54,295 @@ read_column(fitstable_t* tag,
 }
 
     
-/*
- * Implementation for index_t::getCatalog method
- */
 afwTable::SimpleCatalog
 getCatalogImpl(std::vector<index_t*> inds,
-               double ra, double dec, double radius,
-               char const* idcol,
-               std::vector<mag_column_t> const& magcols,
-               char const* stargalcol,
-               char const* varcol,
-               bool unique_ids)
+    lsst::afw::coord::Coord const &ctrCoord,
+    lsst::afw::geom::Angle const &radius,
+    char const* idCol,
+    std::vector<MagColInfo> const& magColInfoList,
+    char const* isStarCol,
+    char const* isVarCol,
+    bool uniqueIds,
+    bool getNewSchema)
 {
     /*
-     If unique_ids == true: return only reference sources with unique IDs;
+     If uniqueIds == true: return only reference sources with unique IDs;
      arbitrarily keep the first star found with each ID.
      */
 
-    size_t const nMag = magcols.size();   /* number of magnitude[error] columns */
-    std::vector<mag_column_t>::const_iterator mc;
+    size_t const nMag = magColInfoList.size();   /* number of magnitude[error] columns */
+    std::vector<MagColInfo>::const_iterator mc;
 
-    for (mc = magcols.begin(); mc != magcols.end(); ++mc) {
-        if (mc->name.size() == 0) {
+    for (mc = magColInfoList.begin(); mc != magColInfoList.end(); ++mc) {
+        if (mc->filterName.size() == 0) {
```

Comment:
Good change. This makes mc local to each for loop, which is a distinct improvement. I'll risk cbegin and hope gcc 4.4 can handle it -- buildbot will tell me (a web search suggested it could, but was not definitive)